tests/functional/adapter/test_seeds_column_types_overrides.py

```diff
@@ -0,0 +1,152 @@
+                }
+            }
+        }
+
+    # everything that goes in the "seeds" directory
+    @pytest.fixture(scope="class")
+    def seeds(self):
+        return {
+            "boolean_type.csv": boolean_type,
+            "datetime_type.csv": datetime_type,
+            "number_type.csv": number_type,
+            "string_type.csv": string_type,
+        }
+
+    def test_seeds_column_overrides(self, project):
+        # seed seeds
+        results = run_dbt(["seed"], expect_pass=True)
+        assert len(results) == 4
+
+        for seed_name, seed_columns in seed_types.items():
+            # retrieve information about columns from trino
+            columns = get_relation_columns(project.adapter, seed_name)
+            for column in columns:
+                (
+                    column_name,
+                    column_data_type,
+                    column_char_size,
+                    column_numeric_precision,
+                    column_numeric_scale,
+                ) = column
+                # if precision/scale/char_length was explicitly specified in column_types config,
+                # compare specified and retrieved column's precision/scale/char_length,
+                # then compare specified and retrieved column type
+                if "(" in seed_columns[column_name]:
+                    # if precision/char_length is specified:
+                    if "," not in seed_columns[column_name]:
+                        (
+                            seed_column_data_type,
+                            seed_column_char_size,
+                            seed_column_data_type_suffix,
+                        ) = re.split(r"[()]", seed_columns[column_name])
+                        # precision/char_length:
+                        assert seed_column_char_size == str(column_char_size)
+                    # if precision and scale are specified:
+                    else:
+                        (
+                            seed_column_data_type,
+                            seed_column_numeric_precision,
+                            seed_column_numeric_scale,
+                            seed_column_data_type_suffix,
+                        ) = re.split(r"[(),]", seed_columns[column_name])
+                        # scale:
+                        assert seed_column_numeric_scale == str(column_numeric_scale)
+                        # precision:
+                        assert seed_column_numeric_precision == str(column_numeric_precision)
+                    # column type:
+                    assert (
+                        "".join([seed_column_data_type, seed_column_data_type_suffix])
+                        == column_data_type
+                    )
+                # if precision/length was NOT explicitly specified,
+                # compare specified and retrieved column type only
+                else:
+                    assert seed_columns[column_name] == column_data_type
```

Comment:
On second thought it would be better to avoid any logic in your tests (the regexes are very similar to the implementation) . Why not add an `expected` field to your seed_types that contains the expected values and compare against that.

Reply:
This approach is fine to me, however it could be extracted to some method and ideally to a Base test class from which other tests could inherit, but it's not in the scope of this PR.

Reply:
@mdesmet those regex may look similar, but they are achieving slightly different goal - in [column.py](https://github.com/starburstdata/dbt-trino/blob/8007a3cd7dab6e4f02a1f867c0788318d08a2cb3/dbt/adapters/trino/column.py#L28) it parses output from [describe](https://github.com/starburstdata/dbt-trino/blob/master/dbt/include/trino/macros/adapters.sql#L8) query run, so actual types of columns. In test, regex is [parsing data in seed_types dictionary](https://github.com/starburstdata/dbt-trino/blob/8007a3cd7dab6e4f02a1f867c0788318d08a2cb3/tests/functional/adapter/test_seeds_column_types_overrides.py#L122), to allow easy comparison for specified and retrieved data types, especially when precision is given.

For example, for `decimal(3,2)` type it would be required to specify 3 expected values:
```python
expected_type = 'decimal'
expected_precision = 3
expected_scale = 2
```

then, we would need to apply some logic anyway, to compare precision/scale if it is expected, unless we want to specify None values for every columns, something like:

```python
expected_type = 'boolean'
expected_char_size = None
expected_precision = None
expected_scale = None
```

Reply:
Fair enough. I'm generally more in favour of explicit testing instead of putting business logic in tests but can definitely live with the proposed solution. I have resolved this conversation.